### DISCUSSION TITLE
Can't containerize an executable  tool

### DISCUSSION THREAD

**Discussion by Fadwa7** [sylabs/singularity#2739] [Q&A]
Hi, 
I'm trying to containerize a tool available on GitHub. I've cloned it using:

```
       git clone --recursive https://github.com/$name/$tool_name.git
        cd $tool_name
        sh install.sh
        cd .. &&
        rm -rf $tool_name

%runscript
        ./$tool_name "$@"


```

The tool is typically installed within a Conda environment. The Conda block works perfectly, but the tool doesn't function when I run:

```
singularity shell --fakeroot --writable $tool_name/

```

Could you please suggest a way to make it executable within Singularity?

Thank you in advance.

Best regards.

**Comment by dtrudg** [comment]
Hello. I've moved your issue to a Q&A discussion as it's a usage question, rather than a bug report or feature request.

It's likely that to get a useful answer you will need to give more detail. Please provide the full definition file you are trying to build, and the command you are using to run the tool. It does matter what tool it is - different tools will install differently, and it's not clear from your post what should be expected.

**Comment by Fadwa7** [comment]
Thank You ! 
Here is the full singularity recipe

```
Bootstrap: docker
From: continuumio/miniconda3:latest

%labels
        authors="Fadwa EL KHADDAR"
        description="Reindeer Image"
%files
	bcalm.yml
%environment
        PATH=/opt/conda/envs/bcalm/bin:$PATH

%post
        apt-get update -y && apt-get install -y build-essential bc wget git unzip
	apt-get install -y cmake-qt-gui
	apt-get install -y zlib1g-dev
        rm -rf /var/lib/apt/lists/*
        mkdir -p /opt/etc
        export DEBIAN_FRONTEND=noninteractive
        export LC_ALL=C.UTF-8
        echo 'export LC_ALL=C.UTF-8' >> "$SINGULARITY_ENVIRONMENT"
        echo 'export LANG=C.UTF-8' >> "$SINGULARITY_ENVIRONMENT" 

        /opt/conda/bin/conda config --append channels default && \
        /opt/conda/bin/conda config --append channels bioconda && \
        /opt/conda/bin/conda config --append channels conda-forge && \
        /opt/conda/bin/conda env create -f bcalm.yml &&
        echo ". /opt/conda/etc/profile.d/conda.sh" >> "$SINGULARITY_ENVIRONMENT"
        echo "source activate bcalm" >> "$SINGULARITY_ENVIRONMENT"           

        git clone --recursive https://github.com/kamimrcht/REINDEER.git
        cd REINDEER
        sh install.sh
	cd .. &&
	rm -rf REINDEER

%runscript
	./Reindeer "$@"
```
I want to make Reindeer executable and when i run the container i want to work with Reindeer, the problem, is that when i run 
	
```
singularity shell --fakeroot --writable $tool_name/
```

bcalm works but not reindeer
